### PR TITLE
Updates Spark ANN benchmark to spark.ml impl

### PR DESCRIPTION
**Pull request by feynmanliang**

### CODE REVIEW

File: spark/spark.scala [hunk 1]
```diff
@@ -1,28 +1,37 @@
...
-val train = MLUtils.loadLibSVMFile(sc, "/mnist.scale")
-val topology = FeedForwardTopology.multiLayerPerceptron(Array[Int](780, 2500, 2000, 1500, 1000, 500, 10), false)
-val trainer = new FeedForwardTrainer(topology, 780, 10).setBatchSize(batchSize)
-trainer.SGDOptimizer.setNumIterations(numIterations).setMiniBatchFraction(1.0).setStepSize(0.03)
+
+val trainData = MLUtils.loadLibSVMFile(sc, "data/mllib/mnist.scale")
+val layers = Array[Int](780, 2500, 2000, 1500, 1000, 500, 10)
+val estimator = new MultilayerPerceptronClassifier()
+  .setLayers(layers)
+  .setBlockSize(128)
```
Comment (avulanov) [inline spark/spark.scala]: Could you extract this parameter as `blockSize` and set it to 10000 as it was previously (see line 11)?

Comment (feynmanliang) [inline spark/spark.scala]: :+1:

File: spark/spark.scala [hunk 2]
```diff
@@ -1,28 +1,40 @@
...
+-val blockSize = 10000
+
+val trainData = MLUtils.loadLibSVMFile(sc, "data/mllib/mnist.scale")
+val layers = Array[Int](780, 2500, 2000, 1500, 1000, 500, 10)
+val estimator = new MultilayerPerceptronClassifier()
+  .setLayers(layers)
+  .setBlockSize(blockSize)
+  .setMaxIter(numIterations)
+  .setSeed(11L)
+
```
Comment (avulanov) [inline spark/spark.scala:25]: The public API of MLP does not allow setting the SGD optimizer. The default optimizer is LBFGS instead. It does not allow direct measuring the throughput as compared to SGD. We should come up with a workaround that allows setting SGD. We might either put the code in the `ann` package, or create a helper class in the private package that will change the optimizer.

Comment (feynmanliang) [inline spark/spark.scala:25]: OK, I agree that that would be useful.

However, I think that is outside of the scope of this PR (which is to fix the benchmark to match the API that was merged). Perhaps we can address that in a separate issue.

**Comment by feynmanliang** [comment]
@avulanov any plans on merging this or should we close this PR?

**Comment by avulanov** [comment]
@feynmanliang This benchmark corresponds to the original branch with a different API https://github.com/avulanov/spark/tree/ann-interface-gemm (as it mentioned on the main page). If we want to fix it to match the API that was merged, we need to find a way to use SGD (that is private in the merged API).

**Comment by feynmanliang** [comment]
Closed because I won't be able to work on this, if someone else wants to take this feel free to cherry pick
